fulp_modules/bloodsuckers/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,746 @@
+			to_chat(owner.current, "<span class='notice'>You prevent your blood from thickening just yet, but you may try again later.</span>")
+			return
+		if(L.blood_volume < level_bloodcost)
+			to_chat(owner.current, "<span class='warning'>You dont have enough blood to thicken your blood, you need [level_bloodcost - L.blood_volume] units more!</span>")
+			return
+		// Buy New Powers
+		var/datum/action/bloodsucker/P = options[choice]
+		AddBloodVolume(-level_bloodcost)
+		BuyPower(new P)
+		to_chat(owner.current, "<span class='notice'>You have used [level_bloodcost] units of blood and learned [initial(P.name)]!</span>")
+	else
+		to_chat(owner.current, "<span class='notice'>You grow more ancient by the night!</span>")
+	/////////
+	// Advance Powers (including new)
+	LevelUpPowers()
+	////////
+	// Advance Stats
+	if(ishuman(owner.current))
+		var/mob/living/carbon/human/H = owner.current
+		var/datum/species/S = H.dna.species
+		S.punchdamagelow += 0.5
+		S.punchdamagehigh += 0.5    // NOTE: This affects the hitting power of Brawn.
+	// More Health
+	owner.current.setMaxHealth(owner.current.maxHealth + 10)
+	// Vamp Stats
+	bloodsucker_regen_rate += 0.05			// Points of brute healed (starts at 0.3)
+	feed_amount += 2				// Increase how quickly I munch down vics (15)
+	max_blood_volume += 100		// Increase my max blood (600)
+	/////////
+	bloodsucker_level ++
+	bloodsucker_level_unspent --
+
+	// Assign True Reputation
+	if(bloodsucker_level == 4)
+		SelectReputation(am_fledgling = FALSE, forced = TRUE)
+	to_chat(owner.current, "<span class='notice'>You are now a rank [bloodsucker_level] Bloodsucker. Your strength, health, feed rate, regen rate, and maximum blood have all increased!</span>")
+	to_chat(owner.current, "<span class='notice'>Your existing powers have all ranked up as well!</span>")
+	update_hud(owner.current)
+	owner.current.playsound_local(null, 'sound/effects/pope_entry.ogg', 25, TRUE, pressure_affected = FALSE)
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+//Assign default team and creates one for one of a kind team antagonists
+
+// Create Objectives
+
+/datum/antagonist/bloodsucker/proc/forge_bloodsucker_objectives() // Fledgling vampires can have different objectives.
+
+	// TEAM
+	//clan = new /datum/team/vampireclan(owner)
+
+	// Lair Objective:		Create a Lair
+	var/datum/objective/bloodsucker/lair/lair_objective = new
+	lair_objective.owner = owner
+	lair_objective.generate_objective()
+	add_objective(lair_objective)
+
+	// Protege Objective
+	var/datum/objective/bloodsucker/protege/protege_objective = new
+	protege_objective.owner = owner
+	protege_objective.generate_objective()
+	add_objective(protege_objective)
+
+	//if (rand(0,1) == 0)
+		// Heart Thief Objective
+	var/datum/objective/bloodsucker/heartthief/heartthief_objective = new
```

Comment:
are you sure that you want ALL fledgeling vampires to want to steal hearts? seems like become kind of a repetitive/stale objective, and one that'd be a pretty big tell for if someone is a vampire ("hey, CMO, can I have some hearts?" *flash spam intensifies*)

Reply:
The other objective is broken and I'm not great at making objectives myself.